pyxem/signals/diffraction_vectors.py

```diff
@@ -97,6 +97,12 @@ def __init__(self, *args, **kwargs):
         _units = kwargs.pop("units", None)
         super().__init__(*args, **kwargs)
         self._set_up_vector(_scales, _offsets, _detector_shape, _column_names, _units)
+        if self._is_object_dtype is None:
+            pass
+        elif self._is_object_dtype:
+            self.ragged = True
+        elif self.ragged == True:
+            self.ragged = False
```

Comment:
This looks weird to me. If the first `elif` clause is met then the second `elif` clause will also be met. If that's by design it's confusing, but I think it's more likely a bug.

Reply:
Let me add some comments and see if I can't clean up the logic. 

That is needed partially because upstream in hyperspy the ragged flag is a bit of a mess....

Sometimes you have data that isn't an object but hyperspy thinks that it is ragged in which case you want to say that it isn't ragged.  Sometimes you have a signal with data=None in the process of initializing the signal in which case you just want to pass through without setting anything and sometimes you have an signal with a object datatype that isn't flagged as ragged....